kodi_addon_checker/versions.py

```diff
@@ -6,13 +6,22 @@
     See LICENSES/README.md for more information.
 """
 
-from packaging.version import parse
+from packaging.version import parse, Version, LegacyVersion
 from kodi_addon_checker import ValidKodiVersions
 
 
 class AddonVersion():
     def __init__(self, version):
-        self.version = parse(str(version))
+        self.version = parse(str(version).lower())
+        # non PEP440 compliant versions (legacy), for beta and alpha versions
+        # convert them into PEP440 format: 1.1.0~beta01 -> 1.1.0beta01
+        if self._islegacy_devversion:
+            self.version = Version(version.replace("~", ""))
+
+    @property
+    def _islegacy_devversion(self):
+        return isinstance(self.version, LegacyVersion) and \
+            ("~beta" or "~alpha" in self.version)
```

Comment:
I was under the impression that you could write whatever after `~` and I probably did at some point? At least for test builds, but I guess that was never supported by core?

Reply:
Core tests only include `beta` and `alpha` so I'm honestly not sure what is supported or not. Note that this validates what we allow on the repo submissions  (and consider best practice) and not exactly what is theoretically possible. If some other edge case arrives, we can always improve on what we have now